migration/migration1.go

```diff
@@ -0,0 +1,137 @@
+package migration
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/query"
+	"github.com/textileio/powergate/ffs"
+	"github.com/textileio/powergate/util"
+)
+
+// V1MultitenancyMigration contains the logic to upgrade a datastore from
+// version 0 to version 1.
+var V1MultitenancyMigration = Migration(func(txn datastore.Txn) error {
+	cidOwners, err := v0CidOwners(txn)
+	if err != nil {
+		return fmt.Errorf("getting cid owners: %s", err)
+	}
+
+	log.Infof("Starting job logger migration...")
+	if err := migrateJobLogger(txn, cidOwners); err != nil {
+		return fmt.Errorf("migrating job logger: %s", err)
+	}
+	log.Infof("Job logger migration finished")
+
+	log.Infof("Starting storage info migration...")
+	if err := migrateStorageInfo(txn, cidOwners); err != nil {
+		return fmt.Errorf("migrating storage info: %s", err)
+	}
+	log.Infof("Storage info migration finished")
+
+	log.Infof("Starting trackstore migration...")
+	if err := migrateTrackstore(txn, cidOwners); err != nil {
+		return fmt.Errorf("migrating trackstore: %s", err)
+	}
+	log.Infof("Trackstore migration finished")
+
+	return nil
+})
+
+func v0CidOwners(txn datastore.Txn) (map[cid.Cid][]ffs.APIID, error) {
```

Comment:
This function (and below ones) are helpers to the sub-migrations to know which APIIDs own Cids.
In many migrations we need to insert data for all the FFS instances that are storing a Cid.